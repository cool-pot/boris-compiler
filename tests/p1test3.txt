*** From outline.txt file

*** Calculator only
print 123*231+(123/45)
*** some variables and calculator
global a=10
print a*a
*** a function
defun square (x) return x*x; end defun
global b=12
print square(a)+square(b)
*** actually, the following is also legal
print square a + square b


*** bubblesort? ***
array arr[1..10];
arr[1]=23;
arr[2]=1;
arr[3]=2;
arr[4]=41;
arr[5]=621;
arr[6]=11;
arr[7]=99;
arr[8]=68;
arr[9]=44;
arr[10]=55;

for i=1..9 do
  for j=2..10 do
     if arr[i]>arr[j] then
        arr[i] <-> arr[j];
     end if
  end for
end for

for i=1..10 to
   print arr[i]
end for   

*** now weird stuff... tuples

tuple t=0,1,3
tuple tt=3,2,1
tuple ttt=0,0,0,0,0,0

ttt = t,tt *** now ttt contains 0,1,3,3,2,1

tt=t *** now tt contains 0,1,3

a[1],a[2]=a[2],a[1] *** exchanged; same as a[1]<->a[2]

*** I am sure we will construct many more interesting examples eventually