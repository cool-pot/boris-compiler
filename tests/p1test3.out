[Token 297] in Line 1, [1,25].
[Token 299] in Line 2, [0,0].
[Token 299] in Line 3, [0,0].
[Token 297] in Line 3, [1,19].
[Token 299] in Line 4, [0,0].
[Token 273] in Line 4, [1,5].
[Token 298] in Line 4, [6,6].
[Token 295] in Line 4, [7,9].With value 123
[Token 292] in Line 4, [10,10].
[Token 295] in Line 4, [11,13].With value 231
[Token 290] in Line 4, [14,14].
[Token 280] in Line 4, [15,15].
[Token 295] in Line 4, [16,18].With value 123
[Token 293] in Line 4, [19,19].
[Token 295] in Line 4, [20,21].With value 45
[Token 281] in Line 4, [22,22].
[Token 299] in Line 5, [0,0].
[Token 297] in Line 5, [1,33].
[Token 299] in Line 6, [0,0].
[Token 261] in Line 6, [1,6].
[Token 298] in Line 6, [7,7].
[Token 296] in Line 6, [8,8].With value 'a'
[Token 282] in Line 6, [9,9].
[Token 295] in Line 6, [10,11].With value 10
[Token 299] in Line 7, [0,0].
[Token 273] in Line 7, [1,5].
[Token 298] in Line 7, [6,6].
[Token 296] in Line 7, [7,7].With value 'a'
[Token 292] in Line 7, [8,8].
[Token 296] in Line 7, [9,9].With value 'a'
[Token 299] in Line 8, [0,0].
[Token 297] in Line 8, [1,14].
[Token 299] in Line 9, [0,0].
[Token 262] in Line 9, [1,5].
[Token 298] in Line 9, [6,6].
[Token 296] in Line 9, [7,12].With value 'square'
[Token 298] in Line 9, [13,13].
[Token 280] in Line 9, [14,14].
[Token 296] in Line 9, [15,15].With value 'x'
[Token 281] in Line 9, [16,16].
[Token 298] in Line 9, [17,17].
[Token 272] in Line 9, [18,23].
[Token 298] in Line 9, [24,24].
[Token 296] in Line 9, [25,25].With value 'x'
[Token 292] in Line 9, [26,26].
[Token 296] in Line 9, [27,27].With value 'x'
[Token 279] in Line 9, [28,28].
[Token 298] in Line 9, [29,29].
[Token 263] in Line 9, [30,32].
[Token 298] in Line 9, [33,33].
[Token 262] in Line 9, [34,38].
[Token 299] in Line 10, [0,0].
[Token 261] in Line 10, [1,6].
[Token 298] in Line 10, [7,7].
[Token 296] in Line 10, [8,8].With value 'b'
[Token 282] in Line 10, [9,9].
[Token 295] in Line 10, [10,11].With value 12
[Token 299] in Line 11, [0,0].
[Token 273] in Line 11, [1,5].
[Token 298] in Line 11, [6,6].
[Token 296] in Line 11, [7,12].With value 'square'
[Token 280] in Line 11, [13,13].
[Token 296] in Line 11, [14,14].With value 'a'
[Token 281] in Line 11, [15,15].
[Token 290] in Line 11, [16,16].
[Token 296] in Line 11, [17,22].With value 'square'
[Token 280] in Line 11, [23,23].
[Token 296] in Line 11, [24,24].With value 'b'
[Token 281] in Line 11, [25,25].
[Token 299] in Line 12, [0,0].
[Token 297] in Line 12, [1,41].
[Token 299] in Line 13, [0,0].
[Token 273] in Line 13, [1,5].
[Token 298] in Line 13, [6,6].
[Token 296] in Line 13, [7,12].With value 'square'
[Token 298] in Line 13, [13,13].
[Token 296] in Line 13, [14,14].With value 'a'
[Token 298] in Line 13, [15,15].
[Token 290] in Line 13, [16,16].
[Token 298] in Line 13, [17,17].
[Token 296] in Line 13, [18,23].With value 'square'
[Token 298] in Line 13, [24,24].
[Token 296] in Line 13, [25,25].With value 'b'
[Token 299] in Line 14, [0,0].
[Token 299] in Line 15, [0,0].
[Token 299] in Line 16, [0,0].
[Token 297] in Line 16, [1,19].
[Token 299] in Line 17, [0,0].
[Token 258] in Line 17, [1,5].
[Token 298] in Line 17, [6,6].
[Token 296] in Line 17, [7,9].With value 'arr'
[Token 277] in Line 17, [10,10].
[Token 295] in Line 17, [11,11].With value 1
[Token 274] in Line 17, [12,13].
[Token 295] in Line 17, [14,15].With value 10
[Token 278] in Line 17, [16,16].
[Token 279] in Line 17, [17,17].
[Token 299] in Line 18, [0,0].
[Token 296] in Line 18, [1,3].With value 'arr'
[Token 277] in Line 18, [4,4].
[Token 295] in Line 18, [5,5].With value 1
[Token 278] in Line 18, [6,6].
[Token 282] in Line 18, [7,7].
[Token 295] in Line 18, [8,9].With value 23
[Token 279] in Line 18, [10,10].
[Token 299] in Line 19, [0,0].
[Token 296] in Line 19, [1,3].With value 'arr'
[Token 277] in Line 19, [4,4].
[Token 295] in Line 19, [5,5].With value 2
[Token 278] in Line 19, [6,6].
[Token 282] in Line 19, [7,7].
[Token 295] in Line 19, [8,8].With value 1
[Token 279] in Line 19, [9,9].
[Token 299] in Line 20, [0,0].
[Token 296] in Line 20, [1,3].With value 'arr'
[Token 277] in Line 20, [4,4].
[Token 295] in Line 20, [5,5].With value 3
[Token 278] in Line 20, [6,6].
[Token 282] in Line 20, [7,7].
[Token 295] in Line 20, [8,8].With value 2
[Token 279] in Line 20, [9,9].
[Token 299] in Line 21, [0,0].
[Token 296] in Line 21, [1,3].With value 'arr'
[Token 277] in Line 21, [4,4].
[Token 295] in Line 21, [5,5].With value 4
[Token 278] in Line 21, [6,6].
[Token 282] in Line 21, [7,7].
[Token 295] in Line 21, [8,9].With value 41
[Token 279] in Line 21, [10,10].
[Token 299] in Line 22, [0,0].
[Token 296] in Line 22, [1,3].With value 'arr'
[Token 277] in Line 22, [4,4].
[Token 295] in Line 22, [5,5].With value 5
[Token 278] in Line 22, [6,6].
[Token 282] in Line 22, [7,7].
[Token 295] in Line 22, [8,10].With value 621
[Token 279] in Line 22, [11,11].
[Token 299] in Line 23, [0,0].
[Token 296] in Line 23, [1,3].With value 'arr'
[Token 277] in Line 23, [4,4].
[Token 295] in Line 23, [5,5].With value 6
[Token 278] in Line 23, [6,6].
[Token 282] in Line 23, [7,7].
[Token 295] in Line 23, [8,9].With value 11
[Token 279] in Line 23, [10,10].
[Token 299] in Line 24, [0,0].
[Token 296] in Line 24, [1,3].With value 'arr'
[Token 277] in Line 24, [4,4].
[Token 295] in Line 24, [5,5].With value 7
[Token 278] in Line 24, [6,6].
[Token 282] in Line 24, [7,7].
[Token 295] in Line 24, [8,9].With value 99
[Token 279] in Line 24, [10,10].
[Token 299] in Line 25, [0,0].
[Token 296] in Line 25, [1,3].With value 'arr'
[Token 277] in Line 25, [4,4].
[Token 295] in Line 25, [5,5].With value 8
[Token 278] in Line 25, [6,6].
[Token 282] in Line 25, [7,7].
[Token 295] in Line 25, [8,9].With value 68
[Token 279] in Line 25, [10,10].
[Token 299] in Line 26, [0,0].
[Token 296] in Line 26, [1,3].With value 'arr'
[Token 277] in Line 26, [4,4].
[Token 295] in Line 26, [5,5].With value 9
[Token 278] in Line 26, [6,6].
[Token 282] in Line 26, [7,7].
[Token 295] in Line 26, [8,9].With value 44
[Token 279] in Line 26, [10,10].
[Token 299] in Line 27, [0,0].
[Token 296] in Line 27, [1,3].With value 'arr'
[Token 277] in Line 27, [4,4].
[Token 295] in Line 27, [5,6].With value 10
[Token 278] in Line 27, [7,7].
[Token 282] in Line 27, [8,8].
[Token 295] in Line 27, [9,10].With value 55
[Token 279] in Line 27, [11,11].
[Token 299] in Line 28, [0,0].
[Token 299] in Line 29, [0,0].
[Token 296] in Line 29, [1,3].With value 'for'
[Token 298] in Line 29, [4,4].
[Token 296] in Line 29, [5,5].With value 'i'
[Token 282] in Line 29, [6,6].
[Token 295] in Line 29, [7,7].With value 1
[Token 274] in Line 29, [8,9].
[Token 295] in Line 29, [10,10].With value 9
[Token 298] in Line 29, [11,11].
[Token 265] in Line 29, [12,13].
[Token 299] in Line 30, [0,0].
[Token 298] in Line 30, [1,1].
[Token 298] in Line 30, [2,2].
[Token 296] in Line 30, [3,5].With value 'for'
[Token 298] in Line 30, [6,6].
[Token 296] in Line 30, [7,7].With value 'j'
[Token 282] in Line 30, [8,8].
[Token 295] in Line 30, [9,9].With value 2
[Token 274] in Line 30, [10,11].
[Token 295] in Line 30, [12,13].With value 10
[Token 298] in Line 30, [14,14].
[Token 265] in Line 30, [15,16].
[Token 299] in Line 31, [0,0].
[Token 298] in Line 31, [1,1].
[Token 298] in Line 31, [2,2].
[Token 298] in Line 31, [3,3].
[Token 298] in Line 31, [4,4].
[Token 298] in Line 31, [5,5].
[Token 266] in Line 31, [6,7].
[Token 298] in Line 31, [8,8].
[Token 296] in Line 31, [9,11].With value 'arr'
[Token 277] in Line 31, [12,12].
[Token 296] in Line 31, [13,13].With value 'i'
[Token 278] in Line 31, [14,14].
[Token 285] in Line 31, [15,15].
[Token 296] in Line 31, [16,18].With value 'arr'
[Token 277] in Line 31, [19,19].
[Token 296] in Line 31, [20,20].With value 'j'
[Token 278] in Line 31, [21,21].
[Token 298] in Line 31, [22,22].
[Token 267] in Line 31, [23,26].
[Token 299] in Line 32, [0,0].
[Token 298] in Line 32, [1,1].
[Token 298] in Line 32, [2,2].
[Token 298] in Line 32, [3,3].
[Token 298] in Line 32, [4,4].
[Token 298] in Line 32, [5,5].
[Token 298] in Line 32, [6,6].
[Token 298] in Line 32, [7,7].
[Token 298] in Line 32, [8,8].
[Token 296] in Line 32, [9,11].With value 'arr'
[Token 277] in Line 32, [12,12].
[Token 296] in Line 32, [13,13].With value 'i'
[Token 278] in Line 32, [14,14].
[Token 298] in Line 32, [15,15].
[Token 283] in Line 32, [16,18].
[Token 298] in Line 32, [19,19].
[Token 296] in Line 32, [20,22].With value 'arr'
[Token 277] in Line 32, [23,23].
[Token 296] in Line 32, [24,24].With value 'j'
[Token 278] in Line 32, [25,25].
[Token 279] in Line 32, [26,26].
[Token 299] in Line 33, [0,0].
[Token 298] in Line 33, [1,1].
[Token 298] in Line 33, [2,2].
[Token 298] in Line 33, [3,3].
[Token 298] in Line 33, [4,4].
[Token 298] in Line 33, [5,5].
[Token 263] in Line 33, [6,8].
[Token 298] in Line 33, [9,9].
[Token 266] in Line 33, [10,11].
[Token 299] in Line 34, [0,0].
[Token 298] in Line 34, [1,1].
[Token 298] in Line 34, [2,2].
[Token 263] in Line 34, [3,5].
[Token 298] in Line 34, [6,6].
[Token 296] in Line 34, [7,9].With value 'for'
[Token 299] in Line 35, [0,0].
[Token 263] in Line 35, [1,3].
[Token 298] in Line 35, [4,4].
[Token 296] in Line 35, [5,7].With value 'for'
[Token 299] in Line 36, [0,0].
[Token 299] in Line 37, [0,0].
[Token 296] in Line 37, [1,3].With value 'for'
[Token 298] in Line 37, [4,4].
[Token 296] in Line 37, [5,5].With value 'i'
[Token 282] in Line 37, [6,6].
[Token 295] in Line 37, [7,7].With value 1
[Token 274] in Line 37, [8,9].
[Token 295] in Line 37, [10,11].With value 10
[Token 298] in Line 37, [12,12].
[Token 296] in Line 37, [13,14].With value 'to'
[Token 299] in Line 38, [0,0].
[Token 298] in Line 38, [1,1].
[Token 298] in Line 38, [2,2].
[Token 298] in Line 38, [3,3].
[Token 273] in Line 38, [4,8].
[Token 298] in Line 38, [9,9].
[Token 296] in Line 38, [10,12].With value 'arr'
[Token 277] in Line 38, [13,13].
[Token 296] in Line 38, [14,14].With value 'i'
[Token 278] in Line 38, [15,15].
[Token 299] in Line 39, [0,0].
[Token 263] in Line 39, [1,3].
[Token 298] in Line 39, [4,4].
[Token 296] in Line 39, [5,7].With value 'for'
[Token 298] in Line 39, [8,8].
[Token 298] in Line 39, [9,9].
[Token 298] in Line 39, [10,10].
[Token 299] in Line 40, [0,0].
[Token 299] in Line 41, [0,0].
[Token 297] in Line 41, [1,29].
[Token 299] in Line 42, [0,0].
[Token 299] in Line 43, [0,0].
[Token 259] in Line 43, [1,5].
[Token 298] in Line 43, [6,6].
[Token 296] in Line 43, [7,7].With value 't'
[Token 282] in Line 43, [8,8].
[Token 295] in Line 43, [9,9].With value 0
[Token 275] in Line 43, [10,10].
[Token 295] in Line 43, [11,11].With value 1
[Token 275] in Line 43, [12,12].
[Token 295] in Line 43, [13,13].With value 3
[Token 299] in Line 44, [0,0].
[Token 259] in Line 44, [1,5].
[Token 298] in Line 44, [6,6].
[Token 296] in Line 44, [7,8].With value 'tt'
[Token 282] in Line 44, [9,9].
[Token 295] in Line 44, [10,10].With value 3
[Token 275] in Line 44, [11,11].
[Token 295] in Line 44, [12,12].With value 2
[Token 275] in Line 44, [13,13].
[Token 295] in Line 44, [14,14].With value 1
[Token 299] in Line 45, [0,0].
[Token 259] in Line 45, [1,5].
[Token 298] in Line 45, [6,6].
[Token 296] in Line 45, [7,9].With value 'ttt'
[Token 282] in Line 45, [10,10].
[Token 295] in Line 45, [11,11].With value 0
[Token 275] in Line 45, [12,12].
[Token 295] in Line 45, [13,13].With value 0
[Token 275] in Line 45, [14,14].
[Token 295] in Line 45, [15,15].With value 0
[Token 275] in Line 45, [16,16].
[Token 295] in Line 45, [17,17].With value 0
[Token 275] in Line 45, [18,18].
[Token 295] in Line 45, [19,19].With value 0
[Token 275] in Line 45, [20,20].
[Token 295] in Line 45, [21,21].With value 0
[Token 299] in Line 46, [0,0].
[Token 299] in Line 47, [0,0].
[Token 296] in Line 47, [1,3].With value 'ttt'
[Token 298] in Line 47, [4,4].
[Token 282] in Line 47, [5,5].
[Token 298] in Line 47, [6,6].
[Token 296] in Line 47, [7,7].With value 't'
[Token 275] in Line 47, [8,8].
[Token 296] in Line 47, [9,10].With value 'tt'
[Token 298] in Line 47, [11,11].
[Token 297] in Line 47, [12,43].
[Token 299] in Line 48, [0,0].
[Token 299] in Line 49, [0,0].
[Token 296] in Line 49, [1,2].With value 'tt'
[Token 282] in Line 49, [3,3].
[Token 296] in Line 49, [4,4].With value 't'
[Token 298] in Line 49, [5,5].
[Token 297] in Line 49, [6,30].
[Token 299] in Line 50, [0,0].
[Token 299] in Line 51, [0,0].
[Token 296] in Line 51, [1,1].With value 'a'
[Token 277] in Line 51, [2,2].
[Token 295] in Line 51, [3,3].With value 1
[Token 278] in Line 51, [4,4].
[Token 275] in Line 51, [5,5].
[Token 296] in Line 51, [6,6].With value 'a'
[Token 277] in Line 51, [7,7].
[Token 295] in Line 51, [8,8].With value 2
[Token 278] in Line 51, [9,9].
[Token 282] in Line 51, [10,10].
[Token 296] in Line 51, [11,11].With value 'a'
[Token 277] in Line 51, [12,12].
[Token 295] in Line 51, [13,13].With value 2
[Token 278] in Line 51, [14,14].
[Token 275] in Line 51, [15,15].
[Token 296] in Line 51, [16,16].With value 'a'
[Token 277] in Line 51, [17,17].
[Token 295] in Line 51, [18,18].With value 1
[Token 278] in Line 51, [19,19].
[Token 298] in Line 51, [20,20].
[Token 297] in Line 51, [21,54].
[Token 299] in Line 52, [0,0].
[Token 299] in Line 53, [0,0].
[Token 297] in Line 53, [1,73].
